### PR TITLE
Kafka homework

### PR DESCRIPTION
**Pull request by nickrafferty78**

### CODE REVIEW

File: kafka-hello-world/src/main/scala/com/labs1904/hwe/homework/KafkaHomework.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    // Create the KafkaConsumer
+    //TODO: Write in a comment what these lines are doing. What are the properties necessary to instantiate a consumer?
+    val properties = getProperties(BootstrapServer)
+    val consumer: KafkaConsumer[String, String] = new KafkaConsumer[String, String](properties)
+
+
+    //TODO: What does this line mean? Write your answer in a comment below
+    consumer.subscribe(Arrays.asList(Topic))
+
+    while ( {
```
Comment (kitmenke) [inline kafka-hello-world/src/main/scala/com/labs1904/hwe/homework/KafkaHomework.scala:33]: Style thing, IMO it should be:
```
while(true) {
...
}
```

File: kafka-hello-world/src/main/scala/com/labs1904/hwe/homework/KafkaHomework.scala [hunk 2]
```diff
@@ -0,0 +1,65 @@
+package com.labs1904.hwe.homework
+
+import java.time.Duration
+import java.util.{Arrays, Properties, UUID}
+
+import net.liftweb.json.DefaultFormats
+import org.apache.kafka.clients.consumer.{ConsumerConfig, ConsumerRecord, ConsumerRecords, KafkaConsumer}
+import org.apache.kafka.common.serialization.StringDeserializer
+
+object KafkaHomework {
```
Comment (kitmenke) [inline kafka-hello-world/src/main/scala/com/labs1904/hwe/homework/KafkaHomework.scala:10]: Do we want to tie this to theory.md somehow? Right now they are two distinct entities

File: kafka-hello-world/src/main/scala/com/labs1904/hwe/homework/theory.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+## Questions
+#### What problem does Kafka help solve? Use a specific use case in your answer 
+* Helpful resource: [Confluent Motivations and Use Cases](https://youtu.be/BsojaA1XnpM)
+
+#### What is Kafka?
+* Helpful resource: [Kafka in 6 minutes](https://youtu.be/Ch5VhJzaoaI) 
+
+#### Describe each of the following with an example of how they all fit together - Topic, Producer, Consumer, Broker, Partition
+
+#### What role does Zookeeper play in Kafka? Is zookeeper being phased out?
```
Comment (kitmenke) [inline kafka-hello-world/src/main/scala/com/labs1904/hwe/homework/theory.md:28]: Might consider cutting this out to focus on core concepts

File: kafka-hello-world/src/main/scala/com/labs1904/hwe/homework/theory.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+3. Commit your changes and push to your forked repository
+
+## Questions
+#### What problem does Kafka help solve? Use a specific use case in your answer 
+* Helpful resource: [Confluent Motivations and Use Cases](https://youtu.be/BsojaA1XnpM)
+
+#### What is Kafka?
+* Helpful resource: [Kafka in 6 minutes](https://youtu.be/Ch5VhJzaoaI) 
+
+#### Describe each of the following with an example of how they all fit together - Topic, Producer, Consumer, Broker, Partition
```
Comment (kitmenke) [inline kafka-hello-world/src/main/scala/com/labs1904/hwe/homework/theory.md:26]: Break this down in to multiple questions, one for each thing?

**Review by kitmenke — COMMENTED** [review]
Some small tweaks, overall looks good

File: kafka-hello-world/src/main/scala/com/labs1904/hwe/homework/KafkaHomework.scala (hunk 2 above)
Comment (nickrafferty78) [reply kafka-hello-world/src/main/scala/com/labs1904/hwe/homework/KafkaHomework.scala:10]: Yea I like that, do you have any ideas on how to tie it in? Ideally we could do it in a way that wasn't too complicated

File: kafka-hello-world/src/main/scala/com/labs1904/hwe/homework/theory.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# Overview
```
Comment (kitmenke) [inline kafka-hello-world/src/main/scala/com/labs1904/hwe/homework/theory.md:1]: Lets move this into the Kafka README and add one step at the end to understand KafkaHomework.scala and attempt to hook it up to our cluster.

**Pull request merged**